### ISSUE TITLE
Apple web push causing errors

### ISSUE THREAD

**Issue by briwg** [web-push-libs/web-push-csharp#97]
Hi,

I'm starting to see registrations for Apple web push and they are causing errors such as:

_Exception thrown: 'System.AggregateException' in mscorlib.dll
An exception of type 'System.AggregateException' occurred in mscorlib.dll but was not handled in user code
One or more errors occurred._

I'm using the latest version of the dll.

Thanks

Brian

**Comment by keke1210** [comment]
@briwg based on the readme file, the Safari browser is not supported. 
![image](https://user-images.githubusercontent.com/44743904/213511450-f7ceb1ea-6692-45c9-977f-f95eb48a79fa.png)
Meanwhile, on the node.js package(https://github.com/web-push-libs/web-push), the safari browser is supported

**Comment by briwg** [comment]
Yes, I understand that. However, the point is that people can land on my website using IOS Safari and select to subscribe to Push Notifications. When I then send out a PN an uncaught exception is thrown by the dll when the apple endpoint is encountered and my sender errors and stops. I have put a workaround in place, but the dll in my view should either a) handle the exception internally b) be updated to support Apple IOS push notifications since it appears they are now finally being rolled out.

**Comment by keke1210** [comment]
@briwg I am rooting for option b.

**Comment by 7702244** [comment]
Safari for MacOS have full support for standard PUSH API which works perfectly with this library. According to presentation, support for iOS and iPadOS will appear next year.

But unfortunately, the library looks abandoned and is no longer being developed.

[WebPush for Safari](https://developer.apple.com/wwdc22/10098)

**Comment by Stratster** [comment]
So i believe i have found the issue here...

After searching through the differences from the main js project and this one, i noticed that this library does not support the content encoding `AES_128_GCM` which is required by safari as you can see in the Authorization header description here:
[Send push notifications to Safari users](https://developer.apple.com/documentation/usernotifications/sending_web_push_notifications_in_safari_and_other_browsers#3994592)

This feature is requested since 2018...: [44](https://github.com/web-push-libs/web-push-csharp/issues/44)

**Comment by briwg** [comment]
DaBeSoft do you know if this works with iOS and iPadOS 16.4  devices which support web push?

See: https://webkit.org/blog/13878/web-push-for-web-apps-on-ios-and-ipados/

**Comment by DaBeSoft** [comment]
Yes, it does work, but afaik the web app needs to be added to the homescreen for Push Notifications to work on iOS / iPadOS (as described in your link).

**Comment by nikolakosturski** [comment]
Any news if the package now support iOS? @DaBeSoft
